### ISSUE TITLE
HTML parser: content gets placed outside the html tag

### ISSUE THREAD

**Issue by akshaykmr080** [remarkjs/react-markdown#320]
The content inside a HTML tag gets placed outside the HTML in the final output

```<a href=\"https://en.wikipedia.org/wiki/HTM\">HTML</a>```
The above HTML gets rendered as 

```<a href=\"https://en.wikipedia.org/wiki/HTM\"></a>HTML```

MY code - 

```

const allowedMarkdownTypes = [
  'paragraph',
  'text',
  'emphasis',
  'strong',
  'link',
  'blockquote',
  'delete',
  'list',
  'listItem',
  'heading',
  'code',
  'thematicBreak',
  'table',
  'tableHead',
  'tableBody',
  'tableRow',
  'tableCell',
  'html',
]

const markdown = "<a href=\"https://en.wikipedia.org/wiki/HTM\">HTML</a>, and will require the <code>html-parser</code> AST plugin to be loaded, in addition to setting the <code>escapeHtml</code> property to false."


<ReactMarkdown
          source={markdown}
          allowedTypes={allowedMarkdownTypes}
          escapeHtml={false}
          linkTarget='_blank'
/>

```

![image](https://user-images.githubusercontent.com/30618712/60204171-1b308800-9803-11e9-919b-05d4850ac029.png)

**Comment by dancrevier** [comment]
Any luck on this? I'm hitting the same issue.

**Comment by dancrevier** [comment]
With a lot of debugging, I figured out what's going on. If you build the sample, you'll see it's got:
`node_modules/unist-visit/node_modules/unist-util-visit-parents`  and from package-lock.json, you can see it's version 2.0.1.
`node_modules/unist-visit-parents` and from pacakge-lock.json, you can see this is version 1.1.2.
If you build the demo app, it includes both and the 2.0.1 wins. In my app, for whatever reason, 1.1.2 won. 1.1.2 doesn't support the reverse parameter and so when parseHtml calls visit, it doesn't do it in the reverse order and things get messed up.

I worked around this problem by adding:
`"unist-util-visit-parents": "^2.0.0"`
to my app's package.json.

I am not enough of an npm expert to know exactly what's going on and what the root cause is. I think react-markdown could work around the problem by adding the same line to the package.json, but there may be a better solution.

**Comment by burgessa23** [comment]
I'm seeing the same thing, de-duping to `"unist-util-visit-parents": "^2.0.0"` doesn't seem to have fixed issues with nested html tags like tables

**Comment by praneybehl** [comment]
Same Issue, content gets placed outside

**Comment by ricvillagrana** [comment]
Same issue `"unist-util-visit-parents": "^2.0.0"` didn't fix the problem for me.

**Comment by emptyOne322** [comment]
same issue

**Comment by NamespaceValentine** [comment]
Having the same issue, forcing  "unist-util-visit-parents": "^2.0.0" did not solve it for me. Neither did forcing "unist-util-visit": "^2.0.0",

**Comment by NamespaceValentine** [comment]
For myself, I fixed this issue by using:
const ReactMarkdown = require("react-markdown/with-html");

It seems that the normal flow can no longer parse html effectively, not sure when/how that happened.

**Comment by emptyOne322** [comment]
> For myself, I fixed this issue by using:
> const ReactMarkdown = require("react-markdown/with-html");
> 
> It seems that the normal flow can no longer parse html effectively, not sure when/how that happened.

not solving issue for me. i have a ssr and everething is fine on server side. But issue apears in browser. Maybee something wrong with bundling with webpack.

**Comment by emptyOne322** [comment]
> Same Issue, content gets placed outside.
> Adding "unist-util-visit-parents": "^2.0.0" to the package.json did fix it for me too.

Solved issue for me

**Comment by h23** [comment]
> > Same Issue, content gets placed outside.
> > Adding "unist-util-visit-parents": "^2.0.0" to the package.json did fix it for me too.
> 
> Solved issue for me

Solved issue for me too

**Comment by ChristianMurphy** [comment]
this appears to be coming from

https://github.com/remarkjs/react-markdown/blob/9b4eb84a78c066a73c1d4a8cb8ed5428baef8e57/package-lock.json#L6662-L6669

which comes from

https://github.com/remarkjs/react-markdown/blob/9b4eb84a78c066a73c1d4a8cb8ed5428baef8e57/package.json#L42

this would need to be updated upstream at

https://gitlab.com/staltz/mdast-add-list-metadata#README

**Comment by wooorm** [comment]
I believe this is fixed on 5.0.0, [with this change](https://github.com/remarkjs/react-markdown/commit/22a5e49ae9dde84257d043f033d1b640b78c7b34#diff-e48a0b02aa7a8f65a453b849e7345933ee0072fb6d0c2a109c83f43cc8fae8c6). Could someone confirm whether that works? Otherwise it’s just the HTML parser being funky and the plan is for that to be replaced next major.

**Comment by wooorm** [comment]
This should be solved by landing GH-563 today, which will soon be released in v6.0.0!
